src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const rightCentury = century
+    ? people.filter(person => Math.ceil(person.died / 100) === century)
+    : people;
+  const menAges = rightCentury
+    .filter(person => person.sex === 'm')
+    .map(person => person.died - person.born);
```

Comment:
Let's improve function performance, why to filter array twice if we can do it only once? Just select needed condition in filter function using ternary operator